articles/tutorials/advanced/2d_shaders/07_sprite_vertex_effect/index.md

```diff
@@ -0,0 +1,863 @@
+
+```csharp
+public VertexPositionColorTexture vertexTL;
+public VertexPositionColorTexture vertexTR;
+public VertexPositionColorTexture vertexBL;
+public VertexPositionColorTexture vertexBR;
+```
+
+
+The [`VertexPositionColorTexture`](xref:Microsoft.Xna.Framework.Graphics.VertexPositionColorTexture) class is a standard MonoGame implementation of the `IVertexType`, and it defines a `Position`, a `Color`, and a `TextureCoordinate` for each vertex. Those should look familiar, because they align with the inputs to the vertex shader function. The alignment is not happenstance, it is enforced by "semantics" that are applied to each field in the vertex. This snippet from the `VertexPositionColorTexture` class defines the semantics for each field in the vertex by specifying the `VertexElementUsage`:
+
+```csharp
+static VertexPositionColorTexture()
+{
+	var elements = new VertexElement[] 
+	{ 
+		new VertexElement(0, VertexElementFormat.Vector3, VertexElementUsage.Position, 0), 
+		new VertexElement(12, VertexElementFormat.Color, VertexElementUsage.Color, 0), 
+		new VertexElement(16, VertexElementFormat.Vector2, VertexElementUsage.TextureCoordinate, 0) 
+	};
+	VertexDeclaration = new VertexDeclaration(elements);
+}
+```
+
+> [!note]
+> MonoGame is open source, so you can go read the full code for [SpriteBatchItem](https://github.com/MonoGame/MonoGame/blob/develop/MonoGame.Framework/Graphics/SpriteBatchItem.cs) and[`VertexPositionColorTexture`](https://github.com/MonoGame/MonoGame/blob/develop/MonoGame.Framework/Graphics/Vertices/VertexPositionColorTexture.cs)
+
+
+The vertex shader declares a semantic for each input using the `:` syntax:
+
+```hlsl
+// the POSITION0 is the semantic 
+float4 position	: POSITION0,
+```
+
+>[!warning] 
+> You cannot change the `SpriteBatch` vertex shader. 
+> 
+> The `SpriteBatch` class does not offer any way to change the vertex semantics that are passed to the shader function.   
+
+### Output Semantics
+
+The same concept of semantics applies to the output of the shader. Here is the output type of the vertex shader function. Notice that the fields also have the `:` semantic syntax. These semantics instruct the graphics pipeline how to use the data:
+
+```hlsl
+struct VSOutput
+{
+	float4 position		: SV_Position;
+	float4 color		: COLOR0;
+    float2 texCoord		: TEXCOORD0;
+};
+```
+
+This is the _input_ struct for the standard pixel shaders from previous chapters. Notice how the fields are named slightly differently, but the _semantics_ are identical:
+
+```hlsl
+struct VertexShaderOutput  
+{  
+   float4 Position : SV_POSITION;  
+   float4 Color : COLOR0;  
+   float2 TextureCoordinates : TEXCOORD0;  
+};
+```
+
+### Matrix Transform
```

Comment:
Might need either a refresher on Worldspace to clip space transition, or an admonition with a link to an external guide to educate the reader.